### PR TITLE
SSH

### PR DESCRIPTION
**Pull request by sthulb**
This is exactly the same as my other SSH PR, I'm just an idiot. :)

### CODE REVIEW

**Comment by ehazlett** [comment]
@sthulb ha np

**Comment by sthulb** [comment]
@ehazlett Care to test?

**Comment by ehazlett** [comment]
@sthulb this needs a rebase please.  thanks!

**Comment by sthulb** [comment]
@ehazlett rebased.

**Comment by ehazlett** [comment]
@sthulb crap i just merged some more before seeing this.  so sorry!  please ping me on irc when you get it so I can take a look before anything else! :)

File: host.go [hunk 1]
```diff
@@ -217,9 +206,9 @@ func (h *Host) StartDocker() error {
 
 	switch h.Driver.GetProviderType() {
 	case provider.Local:
-		cmd, err = h.GetSSHCommand("sudo /etc/init.d/docker start")
+		_, _, err = h.GetSSHCommand("sudo /etc/init.d/docker start")
```
Comment (ehazlett) [inline host.go]: These cannot be a blank identifier as the `cmd.Run()` will be nil.  https://gist.github.com/ehazlett/bbc204d5111d82b3e6b7

Comment (sthulb) [inline host.go]: There is no Run command any more.

Comment (ehazlett) [inline host.go]: Can you explain the Go error in that gist?  It looked like it was trying to Run a nil command I believe.

Comment (ehazlett) [inline host.go]: Specifically https://github.com/sthulb/machine/blob/ssh/host.go#L568

**Comment by sthulb** [comment]
@ehazlett Apparently there were some issues with a rebase...

**Comment by ahmetalpbalkan** [comment]
A big +1 for this! This would improve windows command line prompt (cmd.exe) & PowerShell experience a great deal for users as Windows by default doesn't have ssh.exe in PATH.

**Comment by sthulb** [comment]
So it turns out that writing a SSH client is pretty hard :stuck_out_tongue:

**Comment by ahmetalpbalkan** [comment]
@sthulb hehe that's no surprise at all. I think it makes sense to make this investment here rather than boot2docker, since that is going to go away. So I may take a look into the code as well in the next couple of weeks. :-)

**Comment by sthulb** [comment]
@ahmetalpbalkan I have a fully working client in a branch now.

**Comment by ahmetalpbalkan** [comment]
@sthulb oooh that's sweet would love to try out.

File: host.go [hunk 2]
```diff
@@ -514,46 +473,55 @@ func (h *Host) Provision() error {
...
-	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("error installing docker: %s\n%s\n", err, string(buf.Bytes()))
+		return fmt.Errorf("error installing docker: %s\n%s\n", err, buf.String())
 	}
 
 	return nil
 }
 
-func (h *Host) GetSSHCommand(args ...string) (*exec.Cmd, error) {
+func (h *Host) GetSSHCommand(args ...string) (io.Reader, io.Reader, error) {
```
Comment (ahmetalpbalkan) [inline host.go]: this probably should be named `RunSSHCommand` now as it doesn't get you a cmd anymore + it does the execution.

**Comment by sthulb** [comment]
@ahmetalpbalkan See #909!

**Pull request closed without merging**